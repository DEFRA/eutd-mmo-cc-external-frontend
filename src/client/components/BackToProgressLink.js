import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

const BackToProgressLink = ({ progressUri, documentNumber }) => {
  const { t } = useTranslation();

  return (
    <Link
      id="backToProgress"
      to={progressUri.replace(':documentNumber', documentNumber)}
    >
      {t('commonBackToProgressLinkText')}
      <span className="govuk-visually-hidden">
        {t('commonBackToProgressHiddenText')}
      </span>
    </Link>
  );
};

BackToProgressLink.propTypes = {
  progressUri: PropTypes.string.isRequired,
  documentNumber: PropTypes.string.isRequired,
};

export default BackToProgressLink;