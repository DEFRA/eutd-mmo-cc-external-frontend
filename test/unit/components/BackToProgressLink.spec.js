import * as React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import BackToProgressLink from '../../../src/client/components/BackToProgressLink';

describe('Back to Progress Link', () => {
  const props = {
    documentNumber: 'GBR-23423-4234234',
    progressUri: '/create-catch-certificate/GBR-23423-4234234/progress',
  };
  const { container } = render(
    <MemoryRouter>
      <BackToProgressLink {...props} />
    </MemoryRouter>
  );

  it('should load and render the component', () => {
    expect(screen.getByText('Back to Your Progress')).toBeDefined();
    expect(container).toBeDefined();
  });

  it('should take a snapshot of the component', () => {
    const { container } = render(
      <MemoryRouter>
        <BackToProgressLink {...props} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
