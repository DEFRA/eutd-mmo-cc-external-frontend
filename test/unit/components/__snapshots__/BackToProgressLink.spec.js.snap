// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Back to Progress Link should take a snapshot of the component 1`] = `
<div>
  <a
    href="/create-catch-certificate/GBR-23423-4234234/progress"
    id="backToProgress"
  >
    Back to Your Progress
    <span
      class="govuk-visually-hidden"
    >
      back to your progress page link
    </span>
  </a>
</div>
`;
